File: CarLine.DataCleanUp/Services/Cleanup/BsonValueConverters.cs
using MongoDB.Bson;

namespace CarLine.DataCleanUp.Services.Cleanup;

internal static class BsonValueConverters
{
    public static string ToTrimmedString(BsonValue val)
    {
        return ToStringValue(val).Trim();
    }

    public static string ToStringValue(BsonValue val)
    {
        return val switch
        {
            BsonString s => s.AsString,
            BsonInt32 i => i.ToString(),
            BsonInt64 l => l.ToString(),
            BsonDouble d => d.ToString(),
            BsonBoolean b => b.ToString(),
            _ => val.ToString() ?? string.Empty
        };
    }
}

File: CarLine.DataCleanUp.FSharp/Infrastructure/BsonValueConverters.fs
module CarLine.DataCleanUp.FSharp.BsonValueConverters

open MongoDB.Bson

let toStringValue (value: BsonValue) =
    match value with
    | :? BsonString  as s -> s.AsString
    | :? BsonInt32   as i -> i.ToString()
    | :? BsonInt64   as l -> l.ToString()
    | :? BsonDouble  as d -> d.ToString()
    | :? BsonBoolean as b -> b.ToString()
    | _                   -> value.ToString() |> Option.ofObj |> Option.defaultValue ""

let toTrimmedString (value: BsonValue) =
    toStringValue value |> fun s -> s.Trim()
